common/lib/xmodule/xmodule/capa_module.py

```diff
@@ -193,16 +195,26 @@ def index_dictionary(self):
         """
         Return dictionary prepared with module content and type for indexing.
         """
-        result = super(CapaDescriptor, self).index_dictionary()
-        if not result:
-            result = {}
-        index = {
-            'content_type': self.INDEX_CONTENT_TYPE,
-            'problem_types': list(self.problem_types),
-            "display_name": self.display_name
+        xblock_body = super(CapaDescriptor, self).index_dictionary()
+        # Removing solution
+        capa_content = re.sub(re.compile(r"<solution>.*</solution>", re.DOTALL), "", self.data)
```

Comment:
Thanks for pointing this out. I've added escaping for hints as well as solutions. This regex will remove all solutions, no matter how many of them are in the xml.

I could only find various types of hints, and more general solution tag that contains elements for describing the correct answer.